src/game/common/rts/team.cpp

```diff
@@ -0,0 +1,815 @@
+    m_playerRelations = new PlayerRelationMap();
+    m_teamRelations = new TeamRelationMap();
+
+    if (proto != nullptr) {
+        proto->Prepend_To_Team_Instance_List(this);
+        const TeamTemplateInfo *info = proto->Get_Template_Info();
+
+        if (!info->m_scriptOnAllClear.Is_Empty() || !info->m_scriptOnEnemySighted.Is_Empty()) {
+            m_checkEnemySighted = true;
+        }
+    }
+
+    for (int i = 0; i < ARRAY_SIZE(m_genericScriptActive); i++) {
+        m_genericScriptActive[i] = true;
+    }
+}
+
+Team::~Team()
+{
+    g_theScriptEngine->Notify_Of_Team_Destruction(this);
+
+    for (int i = 0; i < g_thePlayerList->Get_Player_Count(); i++) {
+        Player *player = g_thePlayerList->Get_Nth_Player(i);
+
+        if (player != nullptr) {
+            player->Pre_Team_Destroy(this);
+        }
+    }
+
+    for (;;) {
+        Object *obj = Get_First_Item_In_Team_Member_List();
+
+        if (obj == nullptr) {
+            break;
+        }
+
+        obj->Set_Team(nullptr);
+    }
+
+    captainslog_dbgassert(m_proto != nullptr, "proto should not be null");
+
+    if (m_proto != nullptr && m_proto->Is_In_List_Team_Instance_List(this)) {
+        m_proto->Remove_From_Team_Instance_List(this);
+    }
+
+    m_teamRelations->Delete_Instance();
+    m_playerRelations->Delete_Instance();
+}
+
+void Team::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 1;
+    xfer->xferVersion(&version, 1);
+    unsigned int id = m_teamID;
+    xfer->xferUser(&id, sizeof(id));
+    captainslog_relassert(id == m_teamID, 6, "Team::xfer - TeamID mismatch.  Xfered '%d' but should be '%d'", id, m_teamID);
+    unsigned short size = 0;
+    DLINK_ITERATOR<Object> iter = Iterate_Team_Member_List();
+
+    while (!iter.Done()) {
+        size++;
+        iter.Advance();
+    }
+
+    xfer->xferUnsignedShort(&size);
+
+    if (xfer->Get_Mode() == XFER_SAVE) {
+        DLINK_ITERATOR<Object> iter2 = Iterate_Team_Member_List();
+
+        while (!iter2.Done()) {
+            Object *obj = iter2.Cur();
+            ObjectID id = obj->Get_ID();
+            xfer->xferObjectID(&id);
+            iter2.Advance();
+        }
+    } else {
+        for (int i = 0; i < size; i++) {
+            ObjectID id;
+            xfer->xferObjectID(&id);
+            m_objectIDList.push_back(id);
+        }
+    }
+
+    xfer->xferAsciiString(&m_state);
+    xfer->xferBool(&m_enteredOrExited);
+    xfer->xferBool(&m_active);
+    xfer->xferBool(&m_created);
+    xfer->xferBool(&m_checkEnemySighted);
+    xfer->xferBool(&m_seeEnemy);
+    xfer->xferBool(&m_prevSeeEnemy);
+    xfer->xferBool(&m_wasIdle);
+    xfer->xferInt(&m_destroyThreshold);
+    xfer->xferInt(&m_curUnits);
+    WaypointID waypointid;
+
+    if (m_currentWaypoint != nullptr) {
+        waypointid = m_currentWaypoint->Get_ID();
+    } else {
+        waypointid = WAYPOINT_UNK;
+    }
+
+    xfer->xferUnsignedInt(reinterpret_cast<unsigned int *>(&waypointid));
```

Comment:
Is `WaypointID` int or uint?

Thyme:
```
enum WaypointID : int32_t
```

Reply:
This is what the original code is doing.